nntrainer/graph/network_graph.cpp

```diff
@@ -459,6 +454,7 @@ int NetworkGraph::realizeGraph() {
     }
   }
   /// @todo add check that input_layers <-> output_layers does match.
+  /// @todo check whether graph has a cycle or bipartite
```

Comment:
Could you explain why we need to check it is bipartite?

Reply:
Not related to this pr, but when the multi input and multi output is supported the graph can be bipartite so we need to check it I guess.

Reply:
bipartite graph can be cyclic (bipartite only ensures no cycles of odd length), so you might want to expand on this comment here.

Reply:
Oh no sorry. I mean 2 separated graph not the bipartite. I will amend the comment

Reply:
yes, thanks for the clarification.